### PR TITLE
Enable adapter tests to run on all discovered adapters.

### PR DESCRIPTION
**Pull request by aarongreig**
This is the first step towards having the all the CTS tests parameterize over all available platforms (see #2378).

### CODE REVIEW

**Comment by aarongreig** [comment]
this has an adapter change now so: https://github.com/intel/llvm/pull/16212

**Comment by aarongreig** [comment]
could work without the adapter change included by using the macro from #2376

File: test/conformance/adapter/adapter_adapter_native_cpu.match [hunk 1]
```diff
@@ -1,5 +0,0 @@
-# These pass when the adapter is launched by the loader
```
Comment (RossBrunton) [inline test/conformance/adapter/adapter_adapter_native_cpu.match:1]: Do we want to test this (launching without the loader) any more?

Comment (RossBrunton) [reply test/conformance/adapter/adapter_adapter_native_cpu.match:1]: Ah, saw you just fixed it. Ignore me.

**Review by RossBrunton — APPROVED** [review]

**Review by kbenzie — APPROVED** [review]

**Review by hvdijk — APPROVED** [review]

**Pull request merged**